IdentityCore/src/network/response_serializer/MSIDAADResponseSerializer.m

```diff
@@ -27,11 +27,19 @@
 
 @implementation MSIDAADResponseSerializer
 
-- (id)responseObjectForResponse:(NSHTTPURLResponse *)httpResponse data:(NSData *)data error:(NSError **)error
+- (id)responseObjectForResponse:(NSHTTPURLResponse *)httpResponse
+                           data:(NSData *)data
+                        context:(id <MSIDRequestContext>)context
+                          error:(NSError **)error
 {
-    NSMutableDictionary *jsonObject = [[super responseObjectForResponse:httpResponse data:data error:error] mutableCopy];
+    NSError *jsonError;
+    NSMutableDictionary *jsonObject = [[super responseObjectForResponse:httpResponse data:data context:context error:&jsonError] mutableCopy];
     
-    if (error) { return nil; }
+    if (jsonError)
+    {
+        if (error) *error = jsonError;
+        return nil;
+    }
     
     jsonObject[MSID_OAUTH2_CORRELATION_ID_RESPONSE] = httpResponse.allHeaderFields[MSID_OAUTH2_CORRELATION_ID_REQUEST_VALUE];
```

Comment:
why do we override correlation id? What if there was correlation ID already present in response?

Reply:
after inspecting network traces, I can say that we never get correlation_id in body of response, it always comes in headers as client-request-id